api.go

```diff
@@ -45,6 +68,30 @@ func parseAvailableCarResponse(res *http.Response) (carAvailabilitiesResp, error
 	return parsedBody, err
 }
 
+func GetAvailableFlexCars(ctx context.Context, query CarQuery) (flexCarAvailabilitiesResp, error) {
+	req, err := http.NewRequest(http.MethodGet, flexCarAvailabilityURL, nil)
+	if err != nil {
+		fmt.Printf("client: could not create request: %s\n", err)
+		os.Exit(1)
+	}
+
+	req.WithContext(ctx)
+	q := req.URL.Query()
+	q.Add("BranchID", query.BranchID)
+	q.Add("LanguageID", query.LanguageID)
+	q.Add("CityID", query.CityID)
+	req.URL.RawQuery = q.Encode()
+
+	cars := flexCarAvailabilitiesResp{}
+	res, err := http.DefaultClient.Do(req)
+	if err = json.NewDecoder(res.Body).Decode(&cars); err != nil {
+		test, _ := io.ReadAll(res.Body)
+		fmt.Printf("%s", test)
+		return flexCarAvailabilitiesResp{}, err
+	}
+	return cars, nil
+}
+
 func GetAvailableCars(query CarQuery) ([]Car, error) {
```

Comment:
Should be enhanced with `context` too.

I think we should rename a bunch of function so that we know that it's about "station" cars